### PR TITLE
Issue/10 websockets

### LINKED ISSUE #10: Create lambda websockets
This ticket should be split up as we learn about how this works.

### PR DESCRIPTION
**Pull request by ljoks**
fixes #10 

there is a lambda function separate file for each websocket route:
- connect
- disconnect
- createRoom
- closeRoom
- startQuestion
- endQuestion
- join
- submit

along with a README that describes the input of each route, for use by the mobile and frontend team. 

`dbconnections.js` holds all of the functions that the routes use to query the dynamodb table, which is named Connections. 
while using locally, i installed a package called `serverless-dynamodb-local` to emulate the dynamodb table. 

WIP because the final route still needs to be completed: submit. It requires the QuestionUserResponse routes to first be created so that the user has somewhere to submit their response to.

### CODE REVIEW

File: backend/serverless.yml [hunk 1]
```diff
@@ -99,8 +112,7 @@ functions:
     events:
       - http:
           method: delete
-          path: /api/v1/folder
-
+          path: folder
```
Comment (Goff-Davis) [inline backend/serverless.yml:115]: Is this correct? The other folder endpoints use `/api/v1/folder`

File: backend/src/ws/submit.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Connection } from './dbconnections';
+import {
+	APIGatewayEvent,
+	APIGatewayProxyEvent,
+	Context,
+	ProxyResult,
+} from 'aws-lambda';
+
+let connection: Connection;
+// let room: string;
```
Comment (Goff-Davis) [inline backend/src/ws/submit.ts:10]: This and the below commented variables should be removed if they aren't used.

**Review by Goff-Davis — CHANGES_REQUESTED** [review]
A couple of nit picks

File: backend/docker-compose.yml [hunk 3]
```diff
@@ -13,14 +13,26 @@ services:
       MYSQL_PASSWORD: root
     ports:
       - 3306:3306
-  ucf-poll-redis:
-    image: redis:6
-    volumes:
-      - redis_data:/data
+  # ucf-poll-redis:
```
Comment (kennygperez) [inline backend/docker-compose.yml:16]: Remove commented code

File: backend/src/question_user_response.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { APIGatewayEvent, ProxyResult } from 'aws-lambda';
```
Comment (kennygperez) [inline backend/src/question_user_response.ts:1]: weird path names not very RESTful

**Review by kennygperez — CHANGES_REQUESTED** [review]
Move your ws handlers to a folder called handlers instead

File: backend/src/question_user_response.ts (hunk 4 above)
Comment (ljoks) [reply backend/src/question_user_response.ts:1]: i agree, i was just copying the format Toan used. Not sure what to call it

Comment (kennygperez) [reply backend/src/question_user_response.ts]: well fix it later
 im gonna merge this in for now

**Review by kennygperez — APPROVED** [review]

**Pull request merged**